src/components/bottom_bar_statistic/aggregate_statistics_store.ts

```diff
@@ -0,0 +1,130 @@
+  {
+    name: _t("Max"),
+    types: [CellValueType.number],
+    compute: (values, locale) => max([[values]], locale),
+  },
+  {
+    name: _t("Count"),
+    types: [CellValueType.number, CellValueType.text, CellValueType.boolean, CellValueType.error],
+    compute: (values) => countAny([[values]]),
+  },
+  {
+    name: _t("Count Numbers"),
+    types: [CellValueType.number, CellValueType.text, CellValueType.boolean, CellValueType.error],
+    compute: (values, locale) => countNumbers([[values]], locale),
+  },
+];
+
+export class AggregateStatisticsStore extends SpreadsheetStore {
+  statisticFnResults: StatisticFnResults = {};
+  private isDirty = false;
+
+  constructor(get: Get) {
+    super(get);
+    this.model.selection.observe(this, {
+      handleEvent: this.handleEvent.bind(this),
+    });
+  }
+
+  handle(cmd: Command) {
+    if (
+      (isCoreCommand(cmd) && cmd.type !== "UPDATE_CELL") ||
+      (cmd.type === "UPDATE_CELL" && "content" in cmd)
+    ) {
+      this.isDirty = true;
+    }
+    switch (cmd.type) {
+      case "ACTIVATE_SHEET":
+      case "ACTIVATE_NEXT_SHEET":
+      case "ACTIVATE_PREVIOUS_SHEET":
+      case "EVALUATE_CELLS":
+      case "UNDO":
+      case "REDO":
+        this.isDirty = true;
+    }
+  }
+
+  finalize() {
+    if (this.isDirty) {
+      this.isDirty = false;
+      this.statisticFnResults = this._computeStatisticFnResults();
+    }
+  }
+
+  handleEvent() {
+    if (this.getters.isGridSelectionActive()) {
+      this.statisticFnResults = this._computeStatisticFnResults();
+    }
+  }
+
+  private _computeStatisticFnResults(): StatisticFnResults {
+    const sheetId = this.getters.getActiveSheetId();
+    const cells = new Set<EvaluatedCell>();
+
+    const zones = this.getters.getSelectedZones();
+    for (const zone of zones) {
+      for (const { col, row } of positions(zone)) {
+        if (this.getters.isRowHidden(sheetId, row) || this.getters.isColHidden(sheetId, col)) {
+          continue; // Skip hidden cells
+        }
+
+        const evaluatedCell = this.getters.getEvaluatedCell({ sheetId, col, row });
+        if (evaluatedCell.type !== CellValueType.empty) {
+          cells.add(evaluatedCell);
+        }
+      }
+    }
+
+    const locale = this.getters.getLocale();
+
+    let statisticFnResults: { [name: string]: number | undefined } = {};
+    for (let fn of selectionStatisticFunctions) {
+      // We don't want to display statistical information when there is no interest:
+      // We set the statistical result to undefined if the data handled by the selection
+      // does not match the data handled by the function.
+      // Ex: if there are only texts in the selection, we prefer that the SUM result
+      // be displayed as undefined rather than 0.
+      let fnResult: number | undefined = undefined;
+      const evaluatedCells = [...cells].filter((c) => fn.types.includes(c.type));
+      if (evaluatedCells.length) {
+        fnResult = fn.compute(evaluatedCells, locale);
```

Comment:
we should use lazy values. 99.99% of the time we don't need to compute all the different aggregates

Reply:
up

Reply:
Sorry forgot to post my answer. I doubt that we need to do it as it's really not the costy part. I can investigate a bit further but the cost (changing everything to lazy and lose a bit of readability is such t hat if wze don't see relevant changes in perf, I 'd rather keep things as they are.